utils.py

```diff
@@ -203,9 +203,16 @@ def viewer(sample):
 def prepare(samples):
     print("Preparing data")
 
-    X = []
     y = []
-
+    
+    for sample in samples:
+        image_files = load_imgs(sample)
+        numpics += len(image_files)
+        
+    X = np.empty(shape=(numpics,Sample.IMG_H,Sample.IMG_W,3),dtype=np.uint8)
+    print(f"There are {numpics} samples")
+    
+    idx = 0
     for sample in samples:
```

Comment:
```suggestion
for idx, sample in enumerate(samples):
```

Then we can drop the idx = 0 assignment and the plus at the end of the loop